app/views/viewall/students.html.erb

```diff
@@ -22,10 +55,12 @@
 		<% @neutral = 0 %>
 		<% @registered = 0 %>
 		<tr>
-			<% allties = TieAlumniWithStudentMember.where({ :studentmember_id => "#{student.id}" }) %>
+			<%# allties = TieAlumniWithStudentMember.where({ :studentmember_id => "#{student.id}" }) %>
+			<% allties = @allTies.where({ :studentmember_id => "#{student.id}" }) %>
```

Comment:
@aniarya82 

unfortunately, there is no optimization here. The `where` is interpreted as a query, and there are still the same number of queries. (one query per alumni, and one query per student member, comes to a total of around 1200 queries for one page load, in production).

You will have to find some other way.